Client/Frontend/Browser/Playlist/Cache/HLSThumbnailGenerator.swift

```diff
@@ -0,0 +1,118 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import AVFoundation
+import CoreImage
+import SDWebImage
+
+public class HLSThumbnailGenerator {
+    private enum State {
+        case loading
+        case ready
+        case failed
+    }
+
+    private let asset: AVAsset
+    private let sourceURL: URL
+    private let player: AVPlayer
+    private let videoOutput: AVPlayerItemVideoOutput
+    private var observer: NSKeyValueObservation?
+    private var state: State = .loading
+    private let queue: DispatchQueue
+    private let completion: (UIImage?, TimeInterval?) -> Void
```

Comment:
Maybe error can be added to completion block to indicate the problems stated like 

-  Cannot copy pixel-buffer
-  Failed to seek to specified time
-  Failed to create image from pixel-buffer frame

To add specific error handling, If not please ignore